src/imex.rs

```diff
@@ -106,6 +107,178 @@ pub async fn imex(
     res
 }
 
+pub async fn receive_backup(
+    context: &Context,
+    ticket_bytes: Vec<u8>,
+    passphrase: Option<String>,
+) -> Result<()> {
+    let cancel = context.alloc_ongoing().await?;
+
+    let res = receive_backup_inner(context, ticket_bytes, passphrase.unwrap_or_default())
+        .race(async {
+            cancel.recv().await.ok();
+            Err(format_err!("canceled"))
+        })
+        .await;
+
+    context.free_ongoing().await;
+
+    if let Err(err) = res.as_ref() {
+        // We are using Anyhow's .context() and to show the inner error, too, we need the {:#}:
+        error!(context, "IMEX failed to complete: {:#}", err);
+        context.emit_event(EventType::ImexProgress(0));
+    } else {
+        info!(context, "IMEX successfully completed");
+        context.emit_event(EventType::ImexProgress(1000));
+    }
+
+    res
+}
+
+pub async fn receive_backup_inner(
+    context: &Context,
+    ticket_bytes: Vec<u8>,
+    passphrase: String,
+) -> Result<()> {
+    use iroh_share::{Receiver, Ticket};
+    ensure!(
+        !context.is_configured().await?,
+        "Cannot import backups to accounts in use."
+    );
+    ensure!(
+        context.scheduler.read().await.is_none(),
+        "cannot import backup, IO is running"
+    );
+    let ticket = Ticket::from_bytes(&ticket_bytes)?;
+
+    let recv_dir = tempfile::tempdir().unwrap();
+    let recv_db = recv_dir.path().join("db");
+    let port = 9991;
```

Comment:
Can hardcoding a port become a problem for DeltaChat?  We'd have to have this be reliable on all platforms.

Is there any way we can try out ports until one works and communicate this port to the sender?  Unfortunately it seems the only communication is the QR code, so we can only transfer things from the sender to the receiver currently.

Reply:
Fixed, choosing a random port now.